File: macos/Scanner/Sources/Scanner/Rules/AESCFG004.swift
public struct AESCFG004: Rule {
    public let id = "AES-CFG-004"
    public let severity: Severity = .low

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.configFiles.compactMap { configFile in
            guard configFile.worldReadable else {
                return nil
            }
            return finding(app: configFile.appId, affectedPath: configFile.path)
        }
    }
}

File: windows/Scanner/Rules/AESCFG004.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESCFG004 : IRule
{
    public string Id => "AES-CFG-004";
    public Severity Severity => Severity.Low;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.ConfigFiles
            .Where(configFile => configFile.WorldReadable)
            .Select(configFile => this.Finding(configFile.AppId, affectedPath: configFile.Path))
            .ToArray();
}
